### PR TITLE
Enable logging autests on macOS, clarify why Linux is required

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

File: tests/gold_tests/logging/custom-log.test.py [hunk 1]
```diff
@@ -21,7 +21,8 @@
 Test.Summary = '''
 Test custom log file format
 '''
-# need Curl
+
+# this test depends on Linux specific behavior regarding loopback addrewsses
```
Comment (maskit) [inline tests/gold_tests/logging/custom-log.test.py]: addresses

Comment (randall) [reply tests/gold_tests/logging/custom-log.test.py]: Thanks!

**Comment by randall** [comment]
[approve ci rat]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**